scripts/sonic-kdump-config

```diff
@@ -0,0 +1,633 @@
+    if rc == 0 and type(lines) == list and len(lines) == 0:
+        use_kdump_in_cfg = read_use_kdump()
+        if use_kdump_in_cfg != use_kdump:
+            print_err("Unable to write USE_KDUMP into %s" % kdump_cfg)
+            sys.exit(1)
+    else:
+        print_err("Error while writing USE_KDUMP into %s" % kdump_cfg)
+        sys.exit(1)
+
+## Read current value for KDUMP_NUM_DUMPS in kdump config file
+#
+#  @return The integer value X from KDUMP_NUM_DUMPS=X in /etc/default/kdump-tools
+def read_num_dumps():
+    (rc, lines, err_str) = run_command("grep '#*KDUMP_NUM_DUMPS=.*' %s | cut -d = -f 2" % kdump_cfg, use_shell=True);
+    if rc == 0 and type(lines) == list and len(lines) >= 1:
+        try:
+            return int(lines[0])
+        except Exception as e:
+            print_err('Error! Exception[%s] occured while reading from %s' %(str(e), kdump_cfg))
+            sys.exit(1)
+    else:
+        print_err("Unable to read KDUMP_NUM_DUMPS from %s" % kdump_cfg)
+        sys.exit(1)
+
+## Change the value for KDUMP_NUM_DUMPS in kdump config file /etc/default/kdump-tools
+#
+#  #param num_dumps Integer value for new value
+def write_num_dumps(num_dumps):
+    (rc, lines, err_str) = run_command("/bin/sed -i -e 's/#*KDUMP_NUM_DUMPS=.*/KDUMP_NUM_DUMPS=%d/' %s" % (num_dumps, kdump_cfg), use_shell=False);
+    if rc == 0 and type(lines) == list and len(lines) == 0:
+        num_dumps_in_cfg = read_num_dumps()
+        if num_dumps_in_cfg != num_dumps:
+            print_err("Unable to write KDUMP_NUM_DUMPS into %s" % kdump_cfg)
+            sys.exit(1)
+    else:
+        print_err("Error while writing KDUMP_NUM_DUMPS into %s" % kdump_cfg)
+        sys.exit(1)
+
+## Perform reboot command
+#
+#  We trigger a system reboot, WITHOUT asking for a confirmation
+def perform_reboot():
+    run_command("/usr/bin/config save -y", use_shell=False);
+    (rc, lines, err_str) = run_command("/usr/bin/reboot -y", use_shell=False);
+    if rc == 0 and type(lines) == list and len(lines) == 0:
+        print("We are Rebooting...")
+        os.sleep(60)
+    else:
+        print_err("Error while trying to Reboot")
+        sys.exit(1)
+
+## Command: Enable kdump
+#
+#  @param verbose If True, the function will display a few additinal information
+#  @param yes     If True, will reboot immediately if needed, without comfirmation
+#  @param no      If True, will not display reboot needed message
+#  @return        True is the grub configuration has changed, and False if it has not
+def cmd_kdump_enable(verbose, yes, no):
+
+    kdump_enabled = get_kdump_administrative_mode()
+    memory = get_kdump_memory()
+    num_dumps = get_kdump_num_dumps()
+    if verbose:
+        print("configDB: kdump_enabled=%d memory=[%s] num_nums=%d" % (kdump_enabled, memory, num_dumps))
+
+    current_img = get_current_image();
+    if verbose:
+        print("Current image=[%s]" % current_img)
+    try:
+        lines = [line.rstrip('\n') for line in open(grub_cfg)]
```

Comment:
you cannot assume grub here. sonic also works on aboot. please detect the image type. if it is aboot, then you need to quit.